docs/experimental/examples/oauth2/globus_app.rst

```diff
@@ -1,5 +1,202 @@
 
-Using a Globus App
-==================
+.. _using_globus_app:
 
-TODO
+.. py:currentmodule:: globus_sdk.experimental.globus_app
+
+Using a GlobusApp
+=================
+
+Programmatic communication with Globus services relies on the authorization of requests.
+Management and resolution of this authorization can become an arduous task, especially
+when a script needs to interact with different services each carrying an individual set
+of complex auth requirements. To assist with this task, this library provides a utility
+construct called a GlobusApp.
+
+A :py:class:`~GlobusApp` is a distinct object which will manage auth requirements
+(i.e. **scopes**) identified by their associated service (i.e. **resource server**).
+In addition to storing these requirements, a GlobusApp provides a mechanism to resolve
+unmet ones through browser and api-based auth flows, supplying the resulting tokens to
+bound clients as requested.
+
+There are two flavors of GlobusApp:
+
+*   :py:class:`~UserApp`, a GlobusApp for interactions between an end user and Globus
+    services. Facilitates interactions *on behalf of a user*.
```

Comment:
"On behalf of a user" feels a bit fuzzy, as an application could use a service account to do something on my behalf. One thought:

```suggestion
    services. Operations will be performed as the *user's identity.*.
```